jpyinterpreter-docs/src/modules/ROOT/pages/java-bytecode/java-bytecode.adoc

```diff
@@ -0,0 +1,137 @@
+[[javaBytecode]]
+= Java Bytecode
+
+== Java Virtual Machine Basic Properties
+
+The Java Virtual Machine is a stack based virtual machine.
+It has the following properties:
+
+- Parameters and local variables are stored in slots, with parameters being first.
+This also includes the `this` parameter, which is always in the first slot for instance methods.
+Compilers can also introduce synthetic local variables that do not correspond to any variable in the source.
+
+- On exception, the current stack is cleared and the thrown exception is pushed to the stack.
+
+- Each instruction must have a consistent prior stack size.
+This means if an instruction is the target of multiple instructions, each instruction that have it as a target must have the same post stack size.
```

Comment:
Maybe we want to explain what "being a target for another instruction" means?

Reply:
The example below that is suppose to illustrate what is meant by "being a target for another instruction". Being a target of an instruction means being a possible next instruction for that instruction. Most instructions have their target being the next instruction in the bytecode. Jump instructions usually have two targets: the next instruction in the bytecode, or the instruction to jump to. I can add a definition to the `stack-machine` section where it talks about jumps.